cmake/stm32/utilities.cmake

```diff
@@ -12,18 +12,20 @@ function(stm32_util_create_family_targets FAMILY)
     endif()
 
     if(NOT (TARGET STM32::${FAMILY}${CORE_C}))
+        # Remove unused code, generate thumb code and enable all warnings
+        set(STM32_COMPILE_OPTIONS
+            -mthumb -Wall -ffunction-sections -fdata-sections
+        )
+
         add_library(STM32::${FAMILY}${CORE_C} INTERFACE IMPORTED)
         target_compile_options(STM32::${FAMILY}${CORE_C} INTERFACE 
             --sysroot="${TOOLCHAIN_SYSROOT}"
-            -mthumb -mabi=aapcs -Wall -ffunction-sections -fdata-sections -fno-strict-aliasing -fno-builtin -ffast-math
-            $<$<CONFIG:Debug>:-Og>
-            $<$<CONFIG:Release>:-Os>
+            ${STM32_COMPILE_OPTIONS}
         )
+        # Remove unused code, generate thumb code
         target_link_options(STM32::${FAMILY}${CORE_C} INTERFACE 
             --sysroot="${TOOLCHAIN_SYSROOT}"
-            -mthumb -mabi=aapcs -Wl,--gc-sections
-            $<$<CONFIG:Debug>:-Og>

```

Comment:
I like the -Og for debug builds, why is that bad?

Reply:
It's bad because it overrides Cmake default behavior and default behavior is fine. 
I tested Debug(-g)/Release(-O3)/MinSizeRel(-Os)/ReleaseWithDebInfo(-O2 -g)